clients/python/sliderule/earthdata.py

```diff
@@ -178,6 +178,35 @@ def __cmr_granule_metadata(search_results):
     # - polygons as geodataframe geometry
     return granule_metadata
 
+def __cmr_collection_query(provider, short_name):
+    """Perform a CMR query for collection metadata."""
+    # build params
+    cmr_query_type = 'collections'
+    cmr_format = 'json'
+    CMR_URL = 'https://cmr.earthdata.nasa.gov'
+    cmr_query_url = '/'.join([CMR_URL, 'search', f'{cmr_query_type}.{cmr_format}'])
+    # build list of CMR query parameters
+    params = []
+    params.append(f'?provider={provider}')
+    params.append(f'&short_name={short_name}')
+    # full CMR query url
+    cmr_query_url += "".join(params)
+    logger.debug(f'cmr request={cmr_query_url}\n')
+    # ssl context
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    # query CMR for collection metadata
+    req = urllib.request.Request(cmr_query_url)
+    response = urllib.request.urlopen(req, context=ctx)
```

Comment:
`requests` is a dependency
https://github.com/ICESat2-SlideRule/sliderule/blob/7e16d3e90efc932fc96219a2bba81daf040b44fa/clients/python/requirements.txt#L1

but is urllib easy enough to use instead throughout the client such that requests could be dropped?

Reply:
good point. we might want to create an issue to unify how HTTP requests work